### PR TITLE
complain on missing entries in flutter.yaml

### LINKED ISSUE #3688: Flutter tool crashes if asset is missing
## Observed Behavior

This crash was caused by trying to do "flutter start" when `flutter.yaml` contained an incorrect path to the png file.

## Expected Behavior

The error message should say exactly which file caused the problem (note that the path below doesn't contain the name of the png file.)
Also, the error message should say which area of which file caused the problem (such as the "assets" section in "pubspec.yaml")

## command

flutter start

## exception

FileSystemException: Directory listing failed, path = '/media/jimbe/mojossd/sysui/src/teams/assets/' (OS Error: No such file or directory, errno = 2)

```
dart:io                                                           _Directory.listSync
package:flutter_tools/src/flx.dart 124                            _parseAssets
package:flutter_tools/src/flx.dart 343                            assemble.<async>
===== asynchronous gap ===========================
dart:async                                                        _Completer.completeError
package:flutter_tools/src/flx.dart 419                            assemble.<async>
===== asynchronous gap ===========================
dart:async                                                        Future.Future.microtask
package:flutter_tools/src/flx.dart                                assemble
package:flutter_tools/src/flx.dart 321                            build.<async>
dart:async                                                        _SyncCompleter.complete
package:flutter_tools/src/base/process.dart 60                    runCommandAndStreamOutput.<async>
===== asynchronous gap ===========================
dart:async                                                        _asyncThenWrapperHelper
package:flutter_tools/src/flx.dart                                build
package:flutter_tools/src/flx.dart 253                            buildFlx.<async>
===== asynchronous gap ===========================
dart:async                                                        Future.Future.microtask
package:flutter_tools/src/flx.dart                                buildFlx
package:flutter_tools/src/commands/build_apk.dart 466             buildAndroid.<async>
dart:async                                                        _SyncCompleter.complete
package:flutter_tools/src/commands/build_apk.dart 279             _findApkComponents.<async>
dart:async                                                        _SyncCompleter.complete
package:flutter_tools/src/services.dart 42                        parseServiceConfigs.<async>
===== asynchronous gap ===========================
dart:async                                                        _asyncThenWrapperHelper
package:flutter_tools/src/commands/build_apk.dart                 buildAndroid
package:flutter_tools/src/commands/build_apk.dart 486             buildApk.<async>
===== asynchronous gap ===========================
dart:async                                                        Future.Future.microtask
package:flutter_tools/src/commands/build_apk.dart                 buildApk
package:flutter_tools/src/commands/run.dart 176                   startApp.<async>
===== asynchronous gap ===========================
dart:async                                                        Future.Future.microtask
package:flutter_tools/src/commands/run.dart                       startApp
package:flutter_tools/src/commands/run.dart 124                   RunCommand.runInProject.<async>
===== asynchronous gap ===========================
dart:async                                                        Future.Future.microtask
package:flutter_tools/src/commands/run.dart                       RunCommand.runInProject
package:flutter_tools/src/runner/flutter_command.dart 174         FlutterCommand._run.<async>
dart:async                                                        _SyncCompleter.complete
package:flutter_tools/src/cache.dart 108                          Cache.updateAll.<async>
===== asynchronous gap ===========================
dart:async                                                        _asyncThenWrapperHelper
package:flutter_tools/src/runner/flutter_command.dart             FlutterCommand._run
package:flutter_tools/src/runner/flutter_command.dart 107         FlutterCommand.run
package:args/command_runner.dart 176                              CommandRunner.runCommand.<fn>
dart:async                                                        Future.Future.sync
package:args/command_runner.dart 129                              CommandRunner.runCommand
package:flutter_tools/src/runner/flutter_command_runner.dart 242  FlutterCommandRunner.runCommand
package:args/command_runner.dart 103                              CommandRunner.run.<fn>
dart:async                                                        Future.Future.sync
package:args/command_runner.dart 103                              CommandRunner.run
package:flutter_tools/src/runner/flutter_command_runner.dart 189  FlutterCommandRunner.run
package:flutter_tools/executable.dart 95                          main.<async>.<fn>.<async>
===== asynchronous gap ===========================
package:stack_trace                                               Chain.capture
package:flutter_tools/executable.dart 81                          main.<async>
```

## flutter doctor

```
[✓] Flutter (on Linux, channel master)
    • Flutter at /media/jimbe/mojossd/sysui/src/third_party/flutter
    • Framework revision 60755f6d9c (3 days ago), engine revision 28fbcc3a38

[✓] Android toolchain - develop for Android devices (Android SDK 23.0.2)
    • Android SDK at /usr/local/google/home/jimbe/Android/Sdk
    • Platform android-23, build-tools 23.0.2
    • OpenJDK Runtime Environment (build 1.8.0-google-v7-119990345-117822431)

[✓] Atom - a lightweight development environment for Flutter
    • Atom installed; Flutter plugin version 0.2.2
```

### PR DESCRIPTION
**Pull request by devoncarew**
- complain on missing entries in flutter.yaml (fix https://github.com/flutter/flutter/issues/3688)
- when flutter.yaml validation fails, don't throw an exception; print a message and fail the flx build

W/ missing assets we now print a warning but do complete the flx build. We could also choose to do a hard fail of the build.

@krisgiesing @jason-simmons 

```
[~/flutter/flutter_sunflower] flutter build apk
Building APK in debug mode (android-arm)...
Warning: unable to locate asset entry in flutter.yaml: "lib/bar.png".
Built build/app.apk (8.5MB).
```

### CODE REVIEW

File: packages/flutter_tools/lib/src/flx.dart [hunk 1]
```diff
@@ -50,6 +50,12 @@ class _Asset {
 
   final String source;
 
+  File get assetFile {
+    return new File(source != null ? '$base/$source' : '$base/$relativePath');
+  }
+
+  bool get assertFileExists => assetFile.existsSync();
```
Comment (krisgiesing) [inline packages/flutter_tools/lib/src/flx.dart]: I think this should just be called fileExists if it's not actually asserting anything.

Comment (devoncarew) [inline packages/flutter_tools/lib/src/flx.dart]: Opps, I meant to name this `assetFileExists`.

**Comment by krisgiesing** [comment]
I'm leaning towards a hard failure. I'm trying to imagine a case where as a developer I'd want the build to succeed, but I can't think of one.

File: packages/flutter_tools/lib/src/flx.dart (hunk 1 above)
Comment (krisgiesing) [inline packages/flutter_tools/lib/src/flx.dart]: Hah! OK :)

Comment (devoncarew) [inline packages/flutter_tools/lib/src/flx.dart]: fixed

**Comment by devoncarew** [comment]
I updated this to be a hard failure on a missing asset.

**Comment by krisgiesing** [comment]
LGTM

**Pull request merged**